python/amici/ode_export.py

```diff
@@ -44,6 +44,28 @@
 from .logging import get_logger, log_execution_time, set_log_level
 from .constants import SymbolId
 
+
+# Specialized Pow Function with customized derivative
+class AmiPow(sp.Pow):
+    def _eval_derivative(self, s):
+        dbase = self.base.diff(s)
+        dexp = self.exp.diff(s)
+        part1 = sp.Pow(self.base, self.exp - 1) * self.exp * dbase
+        part2 = self * dexp * sp.log(self.base)
+        if dbase.is_nonzero or self.base.is_nonzero  or dbase.is_nonzero or \
+                part2.is_zero:
+            # first piece never applies or is zero anyways
+            return part1 + part2
+
+        return part1 + sp.Piecewise(
+            (sp.sympify(0.0), sp.And(sp.Eq(self.base, 0), sp.Eq(dbase, 0))),
+            (part2, True)
+        )
+
+    def __repr__(self):
+        return 'AmiPow'
```

Comment:
```suggestion
class CustomPow(sp.Function):
    nargs = (2, )  # function must take 2 arguments

    @classmethod
    def eval(cls, *args):
        return None  # means leave unevaluated

    def _eval_derivative(self, s):
        base, exp = self.args
        dbase = base.diff(s)
        dexp = exp.diff(s)
        part1 = sp.Pow(base, exp - 1) * exp * dbase
        part2 = self * dexp * sp.log(base)
        if dbase.is_nonzero or base.is_nonzero or dbase.is_nonzero or \
                part2.is_zero:
            # first piece never applies or is zero anyways
            return part1 + part2

        return part1 + sp.Piecewise(
            (sp.sympify(0), sp.And(sp.Eq(base, 0), sp.Eq(dbase, 0))),
            (part2, True)
        )
    
    def fdiff(self, argindex=1):
        raise Exception(
            'All derivative calculations should be routed through _eval_derivative()'
        )

    def _eval_is_real(self):
        return True
```

Reply:
This would prevent all kinds of simplifications, such as `powsimp` from working.